packages/styles/src/useStyles.ts

```diff
@@ -0,0 +1,116 @@
+import { RendererContext, ThemeContext } from 'react-fela';
+import { G } from '@mobily/ts-belt';
+import { darken, lighten, transparentize } from 'polished';
+import { breakpointMediaQueries } from './breakpoints';
+import {
+    NativeStyle,
+    NativeStyleObject,
+    NativeStyleOrStylesParam,
+    NativeStyleUtils,
+    Style,
+    StyleOrStylesParam,
+    StyleUtils,
+} from './types';
+import { Theme, NativeTheme } from '@trezor/theme';
+import { multiply, getValueAndUnit, sum } from './utils';
+import { processStyles, processNativeStyles } from './processStyles';
+import { mediaQueries } from './mediaQueries';
+import { useDirection } from './useDirection';
+
+export const useStyles = () => {
+    const renderer = useContext(RendererContext);
+
+    if (!renderer) {
+        throw new Error('The `useStyles()` hook can only be used inside a `StylesProvider`.');
+    }
+
+    const theme = useContext(ThemeContext) as Theme;
+    const nativeTheme = useContext(ThemeContext) as NativeTheme;
+    const directionUtils = useDirection();
+
+    const utils: StyleUtils = useMemo(
+        () => ({
+            ...theme,
+            ...directionUtils,
+            breakpoints: breakpointMediaQueries,
+            darken,
+            getValueAndUnit,
+            lighten,
+            media: mediaQueries,
+            multiply,
+            sum,
+            transparentize,
+        }),
+        [directionUtils, theme],
+    );
+
+    const nativeUtils: NativeStyleUtils = useMemo(
+        () => ({
+            ...nativeTheme,
+            ...directionUtils,
+            darken,
+            lighten,
+            transparentize,
+        }),
+        [directionUtils, nativeTheme],
+    );
+
+    const applyStyle = <TProps extends object>(
+        ...params: Record<string, unknown> extends TProps
+            ? [styleOrStyles: StyleOrStylesParam<TProps>, props?: TProps]
+            : [styleOrStyles: StyleOrStylesParam<TProps>, props: TProps]
+    ) => {
+        const [styleOrStyles, props] = params;
+
+        const invokeStyle = (style: Style<any>) => style(utils, props ?? {});
+
+        const rule = () => {
+            if (G.isArray(styleOrStyles)) {
+                return processStyles(styleOrStyles, invokeStyle);
+            }
+
+            return processStyles([styleOrStyles], invokeStyle);
+        };
+
+        // NOTE: `fela-plugin-bidi` depends on `theme.direction` being set.
+        // NOTE: The default behaviour of `useFela` is to pass `propsWithTheme` down. Our props will
+        // never have a standardized structure, so we don't want any of our plugins to depend on them.
+        return renderer.renderRule(rule as any, { theme: { direction: directionUtils.direction } });
+    };
+
+    const applyNativeStyle = <TProps extends object>(
+        ...params: Record<string, unknown> extends TProps
+            ? [styleOrStyles: NativeStyleOrStylesParam<TProps>, props?: TProps]
```

Comment:
You mean ternary? it's basically `if` for typescript, it will look if you defined type and based on that it will decide if you need to pass props to rule.

```tsx
type MyButtonProps = {
    buttonColor: string;
};

type MyButtonStyleProps = MyButtonProps;

const myButtonStyle = prepareStyle<MyButtonStyleProps>((utils, { buttonColor }) => ({
    cursor: 'pointer',
    width: '100px',
    color: buttonColor,
}));

export const MyButton = ({ buttonColor }: MyButtonProps) => {
    const { applyStyle } = useStyles();

    // Because you used generic MyStyleButtonProps above in prepare style that ternary if will correctly 
    // decide you need also pass this props to applyStyle, other wise you will have to at least always pass empty object
    return <div className={applyStyle(myButtonStyle, { buttonColor })}>Hello world</div>;
};
```